pkg/orchestration/wiring/datadog/types.go

```diff
@@ -15,43 +15,42 @@ const (
 	Memory AlarmType = "memory"
 )
 
-type Alarm struct {
-	Name    string      `json:"name"`
-	Type    string      `json:"type"`
-	Query   string      `json:"query"`
-	Option  AlarmOption `json:"options"`
-	Message string      `json:"message"`
+type AlarmAttributes struct {
+	Name    string       `json:"name"`
+	Type    string       `json:"type"`
+	Query   string       `json:"query"`
+	Options AlarmOptions `json:"options"`
+	Message string       `json:"message"`
 }
 
-type ServiceInstanceSpec struct {
+type OSBInstanceParameters struct {
 	ServiceName voyager.ServiceName `json:"serviceName"`
-	Attributes  Alarm               `json:"attributes"`
+	Attributes  AlarmAttributes     `json:"attributes"`
 	Environment voyager.EnvType     `json:"environment"`
 	Region      voyager.Region      `json:"region"`
 }
-type AlarmOption struct {
-	Timeout           int             `json:"timeout_h,omitempty"`
-	locked            bool            `json:"locked,omitempty"`
-	NotifyNOData      bool            `json:"notify_no_data,omitempty"`
-	NodDataTimeFrame  string          `json:"no_data_timeframe,omitempty"`
-	NotifyAudit       bool            `json:"notify_audit,omitempty"`
-	RequireFullWindow bool            `json:"require_full_window,omitempty"`
-	NewHostDelay      int             `json:"new_host_delay,omitempty"`
-	EscalationMessage string          `json:"escalation_message,omitempty"`
-	RenotifyInterval  int             `json:"renotify_interval,omitempty"`
-	Thresholds        AlarmThresholds `json:"thresholds,omitempty"`
+type AlarmOptions struct {
+	Thresholds        *AlarmThresholds `json:"thresholds,omitempty"`
+	NewHostDelay      *int32           `json:"new_host_delay,omitempty"`
+	RenotifyInterval  *int32           `json:"renotify_interval,omitempty"`
+	TimeoutSeconds    *int32           `json:"timeout_h,omitempty"`
+	EscalationMessage string           `json:"escalation_message,omitempty"`
+	NoDataTimeFrame   string           `json:"no_data_timeframe,omitempty"`
+	Locked            bool             `json:"locked,omitempty"`
+	NotifyNoData      bool             `json:"notify_no_data,omitempty"`
+	NotifyAudit       bool             `json:"notify_audit,omitempty"`
+	RequireFullWindow bool             `json:"require_full_window,omitempty"`
 }
 
 type AlarmThresholds struct {
-	Critical int32 `json:"critical"`
+	Critical int32 `json:"critical,omitempty"`
 	Warning  int32 `json:"warning,omitempty"`
 }
 
 type QueryParams struct {
-	KubeDeployment string
-	KubeNamespace  string
-	Env            string
-	Region         string
-	Threshold      *AlarmThresholds
-	AlarmType      AlarmType
+	KubeDeployment string           `json:"kubeDeployment,omitempty"`
```

Comment:
why some JSON fields are camel cased and others are separated with underscore? e.g. `json:"new_host_delay,omitempty"` above.